src/components/MetaMaskConnection/MetaMaskConnection.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from "react";
+import ConnectButton from "./components/ConnectButton";
+import ConnectedMenu from "./components/ConnectedMenu";
+import { ethereum } from "../../common/types";
+
+async function checkCorrectNetwork(chainId: string) {
+  await ethereum.request({
+    method: "wallet_switchEthereumChain",
+    params: [{ chainId: "0x61" }],
+  }); //todo: if not connected add network
+
+  ethereum.on("chainChanged", () => {
+    window.location.reload();
+  }); // ? remove listener
+}
+
+async function checkWalletConnection(setUserWallet: any) {
+  if (ethereum) {
+    const accounts: string[] = await ethereum.request({
+      method: "eth_requestAccounts",
+    });
+
+    if (accounts.length > 0) {
+      await checkCorrectNetwork(ethereum.chainId);
+
+      const wallet = accounts[0];
+      setUserWallet(wallet);
+
+      return;
```

Comment:
@gavinspr would it be useful for this one to return the wallet incase we want to get that somewhere else in the app and use it quickly not in state

Reply:
I was thinking of putting it in context, and was about to convert the metamask component from chakra to mui. But I'm open to other solutions than context if you had some in mind.